mitmproxy/contentviews/api.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+from abc import ABC, ABCMeta, abstractmethod
+from dataclasses import dataclass
+from typing import ClassVar, Literal
+from mitmproxy import http
+from mitmproxy.flow import Flow
+
+
+class Contentview(ABC):
+    """A contentview that prettifies raw data."""
+
+    @property
+    def name(self) -> str:
+        """
+        The name of this contentview, e.g. "XML/HTML". 
+        Inferred from the class name by default.
+        """
+        return type(self).__name__.removesuffix("Contentview")
+    
+    @abstractmethod
+    def prettify(
+        self,
+        data: bytes,
+        metadata: Metadata,
+    ) -> str:
+        """Transform raw data into human-readable output."""
+
+    def render_priority(
+        self,
+        data: bytes,
+        metadata: Metadata,
+    ) -> float | None:
```

Comment:
Why does this allow `None` and doesn't default to `return 0`?

Reply:
Fixed, thanks.